lib/miam/client.rb

```diff
@@ -266,6 +266,21 @@ def walk_assume_role_policy(role_name, expected_assume_role_policy, actual_assum
     expected_assume_role_policy.sort_array!
     actual_assume_role_policy.sort_array!
 
+    # With only one entity granted
+    # On IAM
+    #   (1) Statement => [ { Principal => AWS => arn } ]
+    # Should be able to specify like:
+    #   (2) Statement => [ { Principal => AWS => [arn] } ]
+    # Actually (1) is reflected when config (2) is applied
+    if expected_assume_role_policy.key?('Statement')
+      expected_assume_role_policy['Statement'].each do |stmt|
+        next unless stmt.key?('Principal')
+        stmt['Principal'].each do |k, v|
+          stmt['Principal'][k] = v[0] if v.is_a?(Array) && v.length < 2
+        end
+      end
+    end
+
```

Comment:
It converts `Principal` hash into an array. We need to convert only values in `Principal` hash.
```diff
Update Role `xxx` > AssumeRolePolicy (dry-run)
 {"Version"=>"2012-10-17",
  "Statement"=>
   [{"Effect"=>"Allow",
-    "Principal"=>{"AWS"=>"arn:aws:iam::NNNNNNNNNNNN:Alice"},
+    "Principal"=>
+     ["AWS", ["arn:aws:iam::NNNNNNNNNNNN:Alice"]],
     "Action"=>"sts:AssumeRole"}]} (dry-run)
```

However your code looks better. How about this?
```ruby
expected_arp_stmt = expected_assume_role_policy.fetch('Statement', [])
expected_arp_stmt = expected_arp_stmt.select {|i| i.key?('Principal') }

expected_arp_stmt.each do |stmt|
  stmt['Principal'].each do |k, v|
    entities = Array(v)
    stmt['Principal'][k] = entities.first if entities.length < 2
  end
end
```